src/app/file/ase_format.cpp

```diff
@@ -1325,6 +1378,142 @@ static void ase_file_write_tileset_chunk(FILE* f, FileOp* fop,
+    size+=4;
+  }
+  else if (std::holds_alternative<std::string>(value)) {
+    const std::string& str = std::get<std::string>(value);
+    ase_file_write_string(f, str);
+    size+=2+str.size();
+  }
+  else if (std::holds_alternative<gfx::Point>(value)) {
+    size+=ase_file_write_point(f, std::get<gfx::Point>(value));
+  }
+  else if (std::holds_alternative<gfx::Size>(value)) {
+    size+=ase_file_write_size(f, std::get<gfx::Size>(value));
+  }
+  else if (std::holds_alternative<gfx::Rect>(value)) {
+    const gfx::Rect& rect = std::get<gfx::Rect>(value);
+    size+=ase_file_write_point(f, rect.origin());
+    size+=ase_file_write_size(f, rect.size());
+  }
+  else if (std::holds_alternative<std::vector<UserData::Variant>>(value)) {
+    const std::vector<UserData::Variant>& v = std::get<std::vector<UserData::Variant>>(value);
+    fputl(v.size(), f);
+    size+=4;
+    const uint16_t type = v.size() == 0 ? 0 : v.front().type();
+    fputw(type, f);
+    size+=2;
+    for (auto elem : v) {
+      size+=ase_file_write_property_value(f, elem);
+    }
+  }
+  else if (std::holds_alternative<UserData::Properties>(value)) {
+    const UserData::Properties& properties = std::get<UserData::Properties>(value);
+    size+=ase_file_write_properties(f, properties);
+  }
+
+  return size;
+}
+
+static long ase_file_write_properties(FILE* f,
+                                      const UserData::Properties& properties)
+{
+  ASSERT(properties.size() > 0);
+
+  long size = 0;
+  fputl(properties.size(), f);
+  size+=4;
+
+  for (auto property : properties) {
+    const std::string& name = property.first;
+    ase_file_write_string(f, name);
+    size+=2+name.size();
+
+    const UserData::Variant& value = property.second;
+    fputw(value.type(), f);
+    size+=2;
+
+    size+=ase_file_write_property_value(f, value);
+  }
+  return size;
+}
+
+static void ase_file_write_properties_maps(FILE* f,
+                                      const dio::AsepriteExternalFiles& ext_files,
+                                      const doc::UserData::PropertiesMaps& propertiesMaps)
+{
+  uint32_t numMaps = propertiesMaps.size();
+  if (numMaps == 0) return;
+
+  long startPos = ftell(f);
+  // Size in bytes of all properties maps stored in this chunk (actual
+  // value is overwritten after serialization of all properties maps)
+  long size = 0;
+  fputl(size, f);
+  fputl(numMaps, f);
+  size+=4;
+  for (auto propertiesMap : propertiesMaps) {
+    const UserData::Properties& properties = propertiesMap.second;
+    // Skip properties map if it doesn't have any property
+    if (properties.size() == 0) continue;
+
+    const std::string& extensionKey = propertiesMap.first;
+    uint32_t extensionId = extensionKey == "" ? 0 : ext_files.to_id.at(extensionKey);
+    fputl(extensionId, f);
+    size+=4;
+
+    size+=ase_file_write_properties(f, properties);
+  }
+  long endPos = ftell(f);
+  fseek(f, startPos, SEEK_SET);
+  fputl(size, f);
```

Comment:
Check that the size could be calculated with a difference between two ftell() (see how it's done with ase_file_write_start_chunk and ase_file_write_close_chunk)